cmd/snap/cmd_run_test.go

```diff
@@ -1972,6 +2031,37 @@ func (s *RunSuite) TestGetSnapDirOptions(c *check.C) {
 	c.Assert(opts, check.DeepEquals, &dirs.SnapDirOptions{HiddenSnapDataDir: true})
 }
 
+func (s *RunSuite) TestRunCompletion(c *check.C) {
+	s.RedirectClientToTestServer(func(w http.ResponseWriter, r *http.Request) {
+		switch r.URL.Path {
+		case "/v2/apps":
+			c.Assert(r.Method, check.Equals, "GET")
+			EncodeResponseBody(c, w, map[string]interface{}{
+				"type":   "sync",
+				"result": fortestingConnectionList,
```

Comment:
I think this needs to return something that the completion logic can iterate through. As it is, this will fail to be deserialized and not exercise the completion logic